types/bell/index.d.ts

```diff
@@ -188,73 +188,110 @@ export interface KnownProviderOptions extends RequiredProviderOptions, OptionalO
  * @param uri the requested resource URI (bell will add the token or authentication header as needed).
  * @param params any URI query parameters (cannot include them in the URI due to signature requirements).
  */
-export type AuthedRequest = (uri: string, params: { [key: string]: string } | null) => Promise<any>;
+export type AuthedRequest = (uri: string, params?: { [key: string]: string }) => Promise<object>;
+
+export interface Credentials {
+  provider: Provider | 'custom';
+  token: string;
+  query: StringLikeMap;
+  /**
+   * Varying data depending on provider.
+   */
+  profile?: object;
+}
+
+export interface Credentials1 extends Credentials {
+  secret: string;
+}
+
+export interface Credentials2 extends Credentials {
+  refreshToken?: string;
+  expiresIn: number;
```

Comment:
expiresIn is a recommended, but not required, field under OAuth 2.0 so may be undefined in some cases.